### ISSUE TITLE
User list on user presence

### ISSUE THREAD

**Issue by gsantopaolo** [IBVSoftware/CollaBoard-Feedback#306]
User presence list 
![image](https://user-images.githubusercontent.com/9937149/30035319-2da704b2-91a9-11e7-8a1f-2993223e27fc.png)

isn't showing correctly users connected to the application.
It is hard to reproduce so this is something that the testing week should solve.
@mtg1986  please organize users so they can work on this bug with @corradocavalli  until is solved
thank you

**Comment by corradocavalli** [comment]
This happens when the app crashes and the user login again, the server return the same user twice because of missing logout. we discussed about implementing some sort of ping message in the past...


On 4 Sep 2017, at 13:43, Gian Paolo Santopaolo <notifications@github.com<mailto:notifications@github.com>> wrote:


Assigned #306<https://github.com/IBVSoftware/CollaBoard-Feedback/issues/306> to @corradocavalli<https://github.com/corradocavalli>.

**Comment by gsantopaolo** [comment]
@corradocavalli no crashes btw did't get user list correctly

**Comment by corradocavalli** [comment]
@gsantopaolo 
There is a bug when there are two or more users logged **with the same account**.

To reproduce:
1: On Machine 1, Log into CB using your MS account
2: Create a new project
3: App receives a login project message with proper machine machine Id (e.g. 01)
4: With user 1 still logged, on machine 2 login **using the same account** and join the same project.
5: Kill CB on machine 1 (stop VS)
6: Close and relaunch CB on machine 2
7: Create a new project
8: Machine 2 receives a LoginProjectMessage message with Machine 1 user Id.

**Comment by gsantopaolo** [comment]
In this particular case having two GP is correct because i was using two pcs with same credential, likewise i do not see the surface hub (JD) that was connected to the same project as well.

About the bug you was talking about when  two or more users logged with the same account.
This is a known issue, for this particular use case, that is going to be solved with a service fabric project that is planned to be developed.

So basically something else we need to test next week

**Comment by corradocavalli** [comment]
You don't see the JD user because the user list is loaded when app, after loading the project, it receives a LoginProjectMessage with machineId that maches the one where project is loading, this basically says: "The server recorded your login into project and you can now safely load the list" we had to use this approach to ensure that when user list is requested it includes also the one currently loading the project.
Since in case of more users logged into project with the same email the message returns the wrong machine Id, the list doesn't get loaded and list goes out of sync.

**Comment by gsantopaolo** [comment]
@corradocavalli there is something very wrong in this feature.
Definitelly need to discuss and reenginer it, thanks

**Comment by corradocavalli** [comment]
Fixed in v109.
